allennlp/data/dataset_readers/dataset_reader.py

```diff
@@ -169,6 +112,18 @@ class DatasetReader(Registrable):
         dataset in each worker, but filtering out the instances that are not needed. If you
         can implement a faster mechanism that only reads part of the data, set this to True,
         and do the sharding yourself.
+    manual_multi_process_sharding : `bool`, optional (default=`False`)
+        The is similar to the `manual_distributed_sharding` parameter, but applies to
+        multi-process data loading. By default, when this reader is used by a multi-process
+        data loader, each worker will filter out all but a subset of the instances
+        that are needed so that you don't end up with duplicates. If you
+        can implement a faster mechanism that only reads part of the data, set this to True,
+        and do the sharding yourself.
```

Comment:
Hmm no I can't think of any reason to. Probably the only scenario where it makes sense to use is if you handle the filtering logic specially within `_read()`, in which case you'd want to set `manual_multi_process_sharding` to `True`.

Maybe we should make this more clear somehow.

Reply:
Maybe I misunderstood, but I think @dirkgr was saying something else: does this automatic sharding actually provide gains if every process is actually reading all of the data?  Would it be better to just disable this functionality until we have a better story?

Reply:
I think we're all on the same page here. There are no gains unless you actually you implement a more efficient way of doing this within `_read()`.

I suggested we remove the `num_workers` parameter before, but @DeNeutoy thought it would be better to keep it so we at least are consistent with the PyTorch API, which I've come to agree with.

Reply:
I agree with being consistent. Let's clarify in the documentation, and maybe even with a warning, that setting `num_workers > 1` and also `manual_multi_process_sharding == False` is safe though useless most of the time. I mainly want to avoid people being confused about this.

Reply:
I just added a warning